frontend/src/components/NewPassword/NewPassword.js

```diff
@@ -22,17 +24,51 @@ class NewPassword extends Component {
     this.setState({ [event.target.name]: event.target.value });
   };
 
+  handleSubmit = (event) => {
+    event.preventDefault();
+    const { token, history } = this.props;
+    const { password, confirmPassword } = this.state;
+    if (password !== confirmPassword) {
+      this.setState({ error: "Passwords don't match" });
+    } else {
+      resetPassword(token, password)
+        .then(() => {
+          history.push('/login');
+        })
+        .catch((error) => {
+          if (error.response.data.errors.reset_password_token[0]) {
+            this.setState({
+              error: `Reset password token ${error.response.data.errors.reset_password_token[0]}.`,
+            });
+          }
+        });
+    }
+  };
+
   render() {
-    const { changeAuthorizedState } = this.props;
-    const { isPasswordVisible } = this.state;
+    const { isPasswordVisible, error } = this.state;
 
+    if (error) {
+      return (
+        <div className="errorMessage">
+          <p>
+            {error}
+            {' '}
+            <Link to="/resetrequest" id="requestLink">
+              Click here to request a new password reset token.
```

Comment:
I might say "You can <Link>request a new password reset token</Link>." I think we can rely on most people knowing that you can click on links. :)

Reply:
@vhoof The reason I worded it that way is because when you look at it on the front end it's not obvious that it's a link. Or at least I didn't think it's obvious since it's the same color text :) I'll change it though. :)

Reply:
Ahh, I'd probably style it like the other links on the page (that blue or whatever).